lib/fs.js

```diff
@@ -376,6 +376,16 @@ fs.exists = function(path, callback) {
   }
 };
 
+Object.defineProperty(fs.exists, internalUtil.promisify.custom, {
+  enumerable: false,
```

Comment:
I think all of the proposed changes here have been (small) improvements or no difference to me. I am to please. :-)

I do think it's about ready to land now.